packages/dependency-extraction-webpack-plugin/index.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * WordPress dependencies
+ */
+const { camelCaseDash } = require( '@wordpress/scripts/utils' );
+
+/**
+ * External dependencies
+ */
+const { createHash } = require( 'crypto' );
+const { ExternalsPlugin } = require( 'webpack' );
+const { RawSource } = require( 'webpack-sources' );
+
+const WORDPRESS_NAMESPACE = '@wordpress/';
+
+class DependencyExtractionWebpackPlugin {
+	constructor( options ) {
+		this.options = Object.assign(
+			{
+				injectPolyfill: false,
+				useDefaults: true,
+			},
+			options
+		);
+		this.externalizedDeps = new Set();
+		this.externalsPlugin = new ExternalsPlugin( 'global', [ this.externalizeWpDeps.bind( this ) ] );
+	}
+
+	externalizeWpDeps( context, request, callback ) {
+		let externRootRequest;
+
+		// Handle via options.requestToExternal first
+		if ( 'function' === typeof this.options.requestToExternal ) {
+			externRootRequest = this.options.requestToExternal( request );
+		}
+
+		// Cascade to default if unhandled and enabled
+		if ( 'undefined' === typeof externRootRequest && this.options.useDefaults !== false ) {
+			externRootRequest = defaultRequestToExternal( request );
+		}
+
+		if ( externRootRequest ) {
+			this.externalizedDeps.add( request );
+			return callback( null, `root ${ externRootRequest }` );
+		}
+
+		return callback();
+	}
+
+	mapRequestToDependency( request ) {
+		// Handle via options.requestToDependency first
+		if ( 'function' === typeof this.options.requestToDependency ) {
```

Comment:
same thing here with yoda conditional... should it be:

```suggestion
		if ( typeof this.options.requestToDependency === 'function' ) {
```